src/main/java/org/broadinstitute/hellbender/utils/pairhmm/SampleSitesForDRAGstrModel.java

```diff
@@ -0,0 +1,437 @@
+                }
+                inStream.close();
+                output.closeArchiveEntry();
+            }
+        }
+    }
+
+    private void saveReferenceDictionary(final ZipArchiveOutputStream output) throws IOException {
+        final SAMSequenceDictionary dictionary = getBestAvailableSequenceDictionary();
+        output.putArchiveEntry(new ZipArchiveEntry("reference.dict"));
+        // need to prevent closing output if the writer's close is called. We simply
+        // flush instead:
+        try (final Writer writer = new OutputStreamWriter(output) {
+                public void close() throws IOException {
+                    flush();
+                }
+        }) {
+            final SAMSequenceDictionaryCodec codec = new SAMSequenceDictionaryCodec(writer);
+            codec.encode(dictionary);
+        }
+        output.closeArchiveEntry();
+    }
+
+    private void composeSummaryText(ZipArchiveOutputStream output) throws IOException {
+        output.putArchiveEntry(new ZipArchiveEntry("summary.txt"));
+        final PrintWriter writer = new PrintWriter(new OutputStreamWriter(output));
+        writer.println("period\trepeat\ttotal\temmitted\tdecimation\tactual_decimation");
+        for (int i = 1; i <= maxPeriod; i++) {
+            for (int j = 1; j <= maxRepeat; j++) {
+                writer.println(String.format("%d\t%d\t%d\t%d\t%.2f\t%.2f",
+                        i, j, totalCounts[i][j], emittedCounts[i][j],
+                        Math.log(decimationTable.decimationMask(i,j) + 1) / Math.log(2),
+                        (- Math.log(emittedCounts[i][j]) + Math.log(totalCounts[i][j])) / Math.log(2) ));
+            }
+        }
+        writer.flush();
+        output.closeArchiveEntry();
+    }
+
+    private static BinaryTableWriter<DragstrLocus> createDragstrLocusWriter(final File outDir, final String name) {
+        try {
+            final File outFile = new File(outDir, name);
+            return DragstrLocus.binaryWriter(outFile);
+        } catch (final IOException  e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    private void traverse(final int seqNumber, final SAMSequenceRecord sequence, final long seqStart, final long seqEnd, final AutoCloseableList<BinaryTableWriter<DragstrLocus>> output, final DecimationTable decimationTable)
+        throws IOException
+    {
+        final String id = sequence.getSequenceName();
+        final NucleotideSequence fullSequence = LazyLoadingReferenceNucleotideSequence.of(directlyAccessEngineReferenceDataSource(), sequence.getSequenceName(),10000);
+        final int maxPeriod = this.maxPeriod;
+        final int length = sequence.getSequenceLength();
+        long pos = seqStart;
+        utter: while (pos <= seqEnd) {
+           final byte base = fullSequence.byteAt(pos);
+           long beg, end, cmp;
+           for (beg = pos - 1; beg >= 1 && fullSequence.byteAt(beg) == base; beg--);
```

Comment:
This could be handled by the stream API, or a while loop. Even if not use {} brackets to encapsulate the beg++; and end--; code.